File: AccuPay/Domain/Payroll/Rate.vb
Option Strict On

Public Class Rate

    Public Property Name As String
    Public Property Rate As Decimal
    Public Property BaseRate As Rate

    Sub New(name As String, rate As Decimal, Optional baseRate As Rate = Nothing)

        Me.Name = name
        Me.Rate = rate
        Me.BaseRate = baseRate

    End Sub

End Class

File: Accupay.Data/ValueObjects/Rate.cs
namespace AccuPay.Data.ValueObjects
{
    public class Rate
    {
        public string Name { get; set; }
        public decimal CurrentRate { get; set; }
        public Rate BaseRate { get; set; }

        public Rate(string name, decimal rate, Rate baseRate = null)
        {
            this.Name = name;
            this.CurrentRate = rate;
            this.BaseRate = baseRate;
        }
    }
}